### PR TITLE
antares: 0.7.24 -> 0.7.28

### PR DESCRIPTION
**Pull request by hatch01**
## Description of changes

Bump antares to 0.7.28 which enables building using electron_30 which is necessary because electron_29 is in EOL.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by emilylange** [comment]
`antares` is currently at version `0.7.24` in nixpkgs and as such a few versions behind.

https://github.com/antares-sql/antares/releases/tag/v0.7.25 bumped `electron` from `29` to `30`.

So I'd suggest updating the package itself to its latest version, at the time of writing https://github.com/antares-sql/antares/releases/tag/v0.7.28, first :)

**Comment by hatch01** [comment]
Vi,
I am working on it.
However I get some errors while updating the package so I will open a new PR when I fixed all of theses

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/an/antares/package.nix [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
   pname = "antares";
-  version = "0.7.24";
+  version = "0.7.28";
 
   src = fetchFromGitHub {
     owner = "antares-sql";
     repo = "antares";
     rev = "v${version}";
-    hash = "sha256-jMtUDqxWwfXl9x61ycohTaacNAhWawL3Z4+OPW5nbOI=";
+    hash = "sha256-zEoEAW68g9694Q284Qt5TCF+zMqxzhYa6GQw+dGnzEg=";
```
Comment (SuperSandro2000) [inline pkgs/by-name/an/antares/package.nix:18]:
```suggestion
    hash = "sha256-nEI1G0A1c+xjALbIcItzh4CFxAeQPOD8h+Bs0aYnEfU=";
```

**Comment by emilylange** [comment]
```
error: hash mismatch in fixed-output derivation '/nix/store/psm1j5l9f408k3f96bq8i2jx6qrcb1cm-antares-0.7.28-npm-deps.drv':
         specified: sha256-3EeHLtqcEnWj70N3nlPHfztzmjVbzdb9f/va4GDRckc=
            got:    sha256-ql7sJHym1QsPPtnndHFHCOeSUcYW51/5FevXGf4v/i4=
```

There is still a hash mismatch.
Also, it would have been better if the `fetchpatch` comment included a link to your unmerged https://github.com/antares-sql/antares/pull/853 PR.
Just so it's clear you aren't trying to hide the fact the commit isn't /actually/ part of the repository yet (but rather GitHub's repository network).

**Comment by hatch01** [comment]
> ```
> error: hash mismatch in fixed-output derivation '/nix/store/psm1j5l9f408k3f96bq8i2jx6qrcb1cm-antares-0.7.28-npm-deps.drv':
>          specified: sha256-3EeHLtqcEnWj70N3nlPHfztzmjVbzdb9f/va4GDRckc=
>             got:    sha256-ql7sJHym1QsPPtnndHFHCOeSUcYW51/5FevXGf4v/i4=
> ```
> 
> There [is still a hash mismatch](https://github.com/NixOS/nixpkgs/pull/335924/checks?check_run_id=29227116358). Also, it would have been better if the `fetchpatch` comment included a link to your unmerged [antares-sql/antares#853](https://github.com/antares-sql/antares/pull/853) PR. Just so it's clear you aren't trying to hide the fact the commit isn't /actually/ part of the repository yet (but rather GitHub's repository network).

I can't use the patch directly from the PR because the PR is going to be merged in develop branch in antares and not in master. So the patch is not applicable.

**Comment by emilazy** [comment]
Best to vendor the patch in Nixpkgs in this case.

**Comment by hatch01** [comment]
ok I will try to do so.
however I face the npm cache error again 
 ```
Installing dependencies
npm error code ENOTCACHED
npm error request to https://registry.npmjs.org/zip-stream failed: cache mode is 'only-if-cached' but no cached response is available.

npm error Log files were not written due to an error writing to the directory: /nix/store/pn3ar164gq31lqky1jjk1f4yga7nprh1-antares-0.7.28-npm-deps/_logs
npm error You can rerun the command with `--loglevel=verbose` to see the logs in your terminal

ERROR: npm failed to install dependencies

Here are a few things you can try, depending on the error:
1. Set `makeCacheWritable = true`
  Note that this won't help if npm is complaining about not being able to write to the logs directory -- look above that for the actual error.
2. Set `npmFlags = [ "--legacy-peer-deps" ]`

```

**Comment by hatch01** [comment]
My last push, build and works on my pc

**Pull request merged**